xbmc/video/VideoDatabase.cpp

```diff
@@ -5432,16 +5466,13 @@ void CVideoDatabase::SetPlayCount(const CFileItem &item, int count, const CDateT
     if (count)
     {
       if (!date.IsValid())
-        strSQL = PrepareSQL("update files set playCount=%i,lastPlayed='%s' where idFile=%i", count, CDateTime::GetCurrentDateTime().GetAsDBDateTime().c_str(), id);
+        strSQL = PrepareSQL("INSERT INTO history(dateWatched, idFile) VALUES ('%s', %i)", CDateTime::GetCurrentDateTime().GetAsDBDateTime().c_str(), id);
       else
-        strSQL = PrepareSQL("update files set playCount=%i,lastPlayed='%s' where idFile=%i", count, date.GetAsDBDateTime().c_str(), id);
+        strSQL = PrepareSQL("INSERT INTO history(dateWatched, idFile) VALUES ('%s', %i)", date.GetAsDBDateTime().c_str(), id);
     }
     else
     {
-      if (!date.IsValid())
-        strSQL = PrepareSQL("update files set playCount=NULL,lastPlayed=NULL where idFile=%i", id);
-      else
-        strSQL = PrepareSQL("update files set playCount=NULL,lastPlayed='%s' where idFile=%i", date.GetAsDBDateTime().c_str(), id);
+        strSQL = PrepareSQL("DELETE FROM history WHERE idFile=%i", id);
```

Comment:
Check this again as the motivation for the four branches wasn't clear.

Reply:
IMO we should never delete the history, only because somebody would like to flag it "unwatched. The reason to mark something unwatched could be a simple as rewatching a TV show or wanting a movie to show up in a certain playlist again. So IMO the playback history should be immutable. Instead we should have a simple "watched" flag per item, which can be unset.

Reply:
While I'm with you I fear that it's going to be very confusing to the user and even to us.
And the whole purpose of this PR is to not change anything for the outside world, so maybe later.

Reply:
I also thought about the possible confusion. But we might be able to come up with a label for this flag that's not/less confusing. Alternatively, we could show a one time dialog when somebody is clicking on "mark as unwatched" that explains what it does. So there are ways to overcome the confusion. But ofc, this can be added later. I just don't see much benefit of the history when you'll still loose it when marking something as unwatched.

Reply:
Well main purpose for me is having multiple watches to sync to trakt for e.g. or sync back from trakt. At the moment you lose all but your last watch per item.
This doesn't address it yet, but is the first step.

Reply:
@da-anda You cannot 'unwatch' or 'unsee' something. The issue here is just nonsense labels. If they're changed to something else e.g "add to watch history" / "delete watch history", it's clear.

Same with the jsonrpc methods: "set playcount" and "set last played" have no meaning when you have a watch history, so just try to interpret what the client is trying to do imo.

Reply:
The "problem" is people trying to rewatch a show normally. Those people normally have everything watched hidden I suppose, so they go and "unwatch" the tvshow to be able to track where they are in their rewatching progress. Normally I get complains about the trakt addon then, as trakt will just overwrite that with it's status.

Reply:
> The "problem" is people trying to rewatch a show normally. Those people normally have everything watched hidden I suppose, so they go and "unwatch" the tvshow to be able to track where they are in their rewatching progress. Normally I get complains about the trakt addon then, as trakt will just overwrite that with it's status.

One word: https://xkcd.com/1172/

Reply:
Exactly, as it is "watched" is gives the users a simple way to flag/tag things, that the UI and (JSON using) remote apps offer easy filtering viw "hide watched". They do not want Kodi to dictate that because it knows they have watched this item they can not change "watched". I don't use the video lib, but I do use the watched flag from fileview

Reply:
@tamland I'm fine with just flushing the history when somebody is clicking "mark unwatched" (or whatever we use as label then) - I just think that this should be a somewhat more "protected" function. At least "hidden" in some "manage" sub-contextmenu and not on same level as "mark watched".

Also, I think, once we have a nice persisted history, we should think about the "watch a TV show again and keep track at which episode you stopped last time" usecase and address it feature wise.